libraries/lib-strings/Base64.cpp

```diff
@@ -0,0 +1,143 @@
+/**********************************************************************
+ 
+ Audacity: A Digital Audio Editor
+ 
+ @file Base64.cpp
+ 
+ @brief Base64 encode/decode (extracted from Audacity sources)
+ 
+ **********************************************************************/
+
+#include "Base64.h"
+#include <wx/string.h>
+
+////////////////////////////////////////////////////////////////////////////////
+// Base64 en/decoding
+//
+// Original routines marked as public domain and found at:
+//
+// http://en.wikibooks.org/wiki/Algorithm_implementation/Miscellaneous/Base64
+//
+////////////////////////////////////////////////////////////////////////////////
+
+// Lookup table for encoding
+const static wxChar cset[] = wxT("ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/");
+const static char padc = wxT('=');
+
+wxString Base64::Encode(const void *in, int len)
+{
+   unsigned char *p = (unsigned char *) in;
+   wxString out;
+
+   unsigned long temp;
+   for (int i = 0; i < len / 3; i++)
+   {
+      temp  = (*p++) << 16; //Convert to big endian
+      temp += (*p++) << 8;
+      temp += (*p++);
+      out += cset[(temp & 0x00FC0000) >> 18];
+      out += cset[(temp & 0x0003F000) >> 12];
+      out += cset[(temp & 0x00000FC0) >> 6];
+      out += cset[(temp & 0x0000003F)];
+   }
+
+   switch (len % 3)
+   {
+      case 1:
+         temp  = (*p++) << 16; //Convert to big endian
+         out += cset[(temp & 0x00FC0000) >> 18];
+         out += cset[(temp & 0x0003F000) >> 12];
+         out += padc;
+         out += padc;
+      break;
+
+      case 2:
+         temp  = (*p++) << 16; //Convert to big endian
+         temp += (*p++) << 8;
+         out += cset[(temp & 0x00FC0000) >> 18];
+         out += cset[(temp & 0x0003F000) >> 12];
+         out += cset[(temp & 0x00000FC0) >> 6];
+         out += padc;
+      break;
+   }
+
+   return out;
+}
+
+int Base64::Decode(const wxString &in, void *out)
+{
+   int len = in.length();
+   unsigned char *p = (unsigned char *) out;
+
+   if (len % 4)  //Sanity check
+   {
+      return 0;
+   }
+
+   int padding = 0;
+   if (len)
+   {
+      if (in[len - 1] == padc)
+      {
+         padding++;
+      }
+
+      if (in[len - 2] == padc)
+      {
+         padding++;
+      }
+   }
+
+   //const char *a = in.mb_str();
+   //Setup a vector to hold the result
+   unsigned long temp = 0; //Holds decoded quanta
+   int i = 0;
+   while (i < len)
+   {
+      for (int quantumPosition = 0; quantumPosition < 4; quantumPosition++)
```

Comment:
`size_t`

Reply:
Not used as an index, but rather a loop iteration counter